README.md

```diff
@@ -80,6 +81,16 @@ empty_array_mt
 --------------
 **syntax:** `setmetatable({}, cjson.empty_array_mt)`
 
+**NOTE**: Use of this field is *discouraged* as of the `2.1.0.6` release. This
+field is considered *depcrecated*, and has been renamed to
+[array_mt](#array_mt) for expliciteness.
+
+[Back to TOC](#table-of-contents)
+
+array_mt
+--------
+**syntax:** `setmetatable({}, cjson.array_mt)`
+
 A metatable which can "tag" a table as a JSON Array in case it is empty (that is, if the
```

Comment:
Maybe the documentation wording is not clear enough. What cjson does about it is here:

https://github.com/openresty/lua-cjson/blob/master/lua_cjson.c#L716-L724

As we can see from that logic, it deals with ambiguities when a table has a 0 length:

  * if cjson is encoding a table:
    * that table has `len > 0` -> array `[...]`
    * table has `len == 0`
      * if `!cfg.encode_empty_table_as_object` -> array `[]`
      * if table's metatable is the `cjson_empty_array` -> array `[]`
      * else -> object `{}`

The length of the table comes from [this function](https://github.com/openresty/lua-cjson/blob/master/lua_cjson.c#L520) (link reference only). It seems to simply rely on the presence of non-integer keys yielded by `lua_next()` to determine if a table is an Object/Array.

What I think I understand from your use case, is that is it about having another metatable, one that if set on a table that has both a hash and an array part (it would fail the current check from `lua_array_length` because it has non-integer keys), then still *force* the encoding of the array part only, as a table. Is that correct?